### PR TITLE
Added default implementation of IRequestHandler

### PR DESCRIPTION
**Pull request by martinekvili**
This PR was requested by @amaitland in issue #1437, but as far as I was able to tell, it was never implemented.

### CODE REVIEW

**Comment by perlun** [comment]
@martinekvili Thanks for the PR. Could you resolve the conflicts? We can then review it.

**Comment by martinekvili** [comment]
@perlun Thanks, the conflicts should be sorted now.

**Comment by perlun** [comment]
@amaitland Since you requested the original change, I think it would make most sense if you review this. Let me know if you are unable to do this and we'll let someone else give it a look.

**Comment by amaitland** [comment]
@perlun Seems reasonable, will have a look 👍

File: CefSharp/DefaultRequestHandler.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
+﻿// Copyright © 2010-2017 The CefSharp Authors. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license that can be found in the LICENSE file.
+
+using System.Security.Cryptography.X509Certificates;
+
+namespace CefSharp
```
Comment (amaitland) [inline CefSharp/DefaultRequestHandler.cs]: At a minimum this should be in the `Handler` folder. Probably the `CefSharp.Handler` namespace as well. (Ideally all the handlers should be in their own namespace, moving the interfaces is a massive breaking change, so probably won't happen).

File: CefSharp/DefaultRequestHandler.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+{
+    /// <summary>
+    /// Default implementation of <see cref="IRequestHandler"/>.
+    /// This class provides default implementations of the methods from <see cref="IRequestHandler"/>,
+    /// therefore providing a convenience base class for any custom request handler.
+    /// </summary>
+    public class DefaultRequestHandler : IRequestHandler
+    {
+        public virtual bool GetAuthCredentials(IWebBrowser browserControl, IBrowser browser, IFrame frame, bool isProxy, string host,
```
Comment (amaitland) [inline CefSharp/DefaultRequestHandler.cs]: Ideally the `xmldoc` would be copied from the parent `IRequestHandler`. `GhostDoc` is handy when it comes to this sort of thing.

**Comment by martinekvili** [comment]
@amaitland Thank you very much for the feedback! I think I implemented all the requested changes.

**Comment by perlun** [comment]
@amaitland AppVeyor has stalled on this one for some reason, but I'm fine with you merging this nonetheless.

**Pull request merged**